tests/phpunit/ConfigDefinitionTest.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\ProcessorIconv\Tests;
+
+use Keboola\Component\BaseComponent;
+use Keboola\ProcessorIconv\ConfigDefinition;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Config\Definition\Exception\InvalidConfigurationException;
+use Symfony\Component\Config\Definition\Processor;
+
+class ConfigDefinitionTest extends TestCase
+{
+    public function testPass() : void
+    {
+        $sourceConfig = [
+            'parameters' => [
+                'source_encoding' => 'UTF-8',
+            ],
+        ];
+        $targetConfig = [
+            'parameters' => [
+                'source_encoding' => 'UTF-8',
+                'target_encoding' => 'UTF-8',
+                'on_error' => 'fail',
+            ],
+        ];
+        $processor = new Processor();
+        $cfgDefinition = new ConfigDefinition();
+        $processedConfig = $processor->processConfiguration($cfgDefinition, [$sourceConfig]);
+        self::assertEquals($targetConfig, $processedConfig);
+    }
+
+
+    /**
+     * @dataProvider invalidConfigurationProvider
+     * @param array $configuration
+     * @param string $message
+     */
+    public function testFail(array $configuration, string $message) : void
+    {
+        putenv('KBC_DATADIR=' . __DIR__ . '/../functional/basic/source/data');
```

Comment:
uf :/ Aha, tak jsem to přečetl moc rychle. A moc se mi to nelíbí. Proč ne 
```
$sourceConfig = [
            'parameters' => [
                'source_encoding' => 'does not exist',
            ],
        ];
``` a jen udělat config z toho pole?

Reply:
Safra, nejdřív číst a pak psát ....

Reply:
Ale myslím, že tam není potřeba vytvářet celou tu komponentu, ne?

Reply:
Jasně... užžžžž sorry. Ono to nastavuje ten error handler!

Reply:
To by chtělo asi nějak nastavit, abys nemusel postavit celý vesmír jen abys zavolal setEnvironment... udělat ho static?

Reply:
taky me to napadlo, ale nechci se tim ted zabyvat, beru to, ze je to takovej edgecase

Reply:
https://github.com/keboola/php-component/issues/22